### ISSUE TITLE
Entire cards should not be clickable

### ISSUE THREAD

**Issue by chlane** [vmware/clarity#4579]
[Clarity Version]
3.1.1

[Test URL]
https://clarity.design/documentation/cards#guidelines

[WCAG 2.1 Guideline]
Success Criterion 4.1.2 Name, Role, Value
(Level A)

[OS]
any

[User Agents/Assistive Technologies]
any

[Component]
Cards

[Description]
Entire cards should not be clickable. The current documentation promotes clickable cards as seen below.

&quot;Clickable cards be predictable
When the entire card is clickable, the resulting action must be an expected outcome. A common action is to navigate to more details. Not every card need be clickable.&quot;

[User Impact]
Assistive technologies will have difficulty composing accessible names for cards that are implemented as links or buttons. Focus order issues may occur as well since buttons or links are not meant to contain interactive child elements.

[Steps to Reproduce]
- Go to the Test URL and see the text promoting clickable cards

[Recommendation]
The documentation should not recommend making entire cards clickable

**Comment by sdurnov** [comment]
Clickable cards and areas are widely used by many people. For instance, even Apple wraps entire cards/blocks in <figure>, and then in <a>, which makes whole area clickable, but keeps other clickable elements inside still independently accessible. Check apple.com home page layout blocks as the example.
As for the Cards, again, Clarity card is pretty usable without any stock card content. I often use it with my own custom content. And it may not even have any clickable elements inside, just being a clickable element itself.
Also, when I read the docs, I didn't have any perception that it somehow promotes making cards clickable. It looks clear that this is just an edge case, needed rarely, and is not promoted as a default behavior. I'd keep it as it is or just add a warning to docs that such behavior may not be accessible.

**Comment by chlane** [comment]
@sdurnov If they are widely used as you mentioned they are not an edge case and will create extremely long and unusable link names for blind screen reader users. This should be avoided.

**Comment by sdurnov** [comment]
> @sdurnov If they are widely used as you mentioned they are not an edge case and will create extremely long and unusable link names for blind screen reader users. This should be avoided.

I agree that it should be avoided. Btw it would be nice to keep such features in docs, while marking it with some kind of a low accessibility badge.

**Comment by chlane** [comment]
@sdurnov Hi Surgey,

Another issue is that when Clarity consuming developers add links inside the clickable card's content they well get HTML validation errors. 

For example when validating the following HTML at https://validator.w3.org/nu/#textarea.

```
<a _ngcontent-clarity-c223="" href="javascript://" class="card clickable"><div _ngcontent-clarity-c223="" class="card-img"><img _ngcontent-clarity-c223="" src="assets/images/documentation/cards/placeholder_350x150.png"></div><div _ngcontent-clarity-c223="" class="card-block"><p _ngcontent-clarity-c223="" class="card-text"> Lorem ipsum dolor sit amet, consectetur adipisicing elit. Adipisci consectetur magnam eos amet sit rem. Ipsam maiores incidunt eum quasi enim! Corporis sunt nisi totam molestias quam commodi maxime mollitia. </p></div>
<a href="#">test link</a>
</a>
```
these errors occur.
```
Error: Start tag a seen but an element of the same type was already open
Error: Stray end tag a.
```
HTML parsing errors violate the following WCAG Success Criterion and can cause unpredictable issues for users of assistive technologies.

Success Criterion 4.1.1 Parsing
(Level A)
https://www.w3.org/TR/WCAG21/#parsing

I strongly recommend against using or promoting clickable cards regardless of demand or potential 3rd party solutions, which will be inaccessible anyway.

**Comment by sdurnov** [comment]
@chlane good point! But still, I would add this info into UI guidelines for cards as non-recommended use instead of completely removing it from docs. This will effectively inspire at least some developers to reconsider the UI solution instead of looking for a workaround or a 3rdparty solution.

**Comment by chlane** [comment]
@sdurnov Hi Sergey, I like that. I recommend that we teach them not to do it ;)

**Comment by chlane** [comment]
Amy Li (amyli@vmwarecom) on the A11Y team found a great solution for this at http://ebay.github.io/mindpatterns/navigation/tile/index.html

**Comment by mathisscott** [comment]
@chlane 
That example has nested anchor links. While the link hrefs go to the same place, it appears to violate our guidance of not putting clickable items inside of other clickable items. 

In addition, those tiles from ebay are only anchor links. Product teams want clickable cards that act as both buttons and radio buttons too. I don't know how this recommendation moves us towards that.

**Comment by chlane** [comment]
@mathisscott Amy LI can explain but she needs access to this thread. Her github user name is @amylinow and her email address is amyli@vmwarecom

**Comment by mathisscott** [comment]
The reason Amy can't comment on this thread is because it is closed. I recommend opening a new issue if this conversation needs to continue.

CC: @gnomeontherun
